rust/hyperlane-core/src/traits/indexer.rs

```diff
@@ -20,18 +21,23 @@ pub enum IndexMode {
     #[default]
     Block,
     /// Sequence based indexing.
-    Sequence,
+    MessageNonce,
 }
 
+/// An inclusive range of sequence numbers.
+pub type SequenceRange = RangeInclusive<u32>;
+
 /// An indexing range.
-#[derive(Copy, Debug, Clone)]
+#[derive(Debug, Clone)]
 pub enum IndexRange {
     /// For block-based indexers
-    Blocks(u32, u32),
+    BlockRange(SequenceRange),
     /// For indexers that look for specific sequences, e.g. message nonces.
-    Sequences(u32, u32),
+    MessageNonceRange(SequenceRange),
```

Comment:
Because this isn't just for outbound message indexing (i.e. we want to index processed messages based off a counter that's increased for each processed message, & we want something similar for IGP payments), imo SequenceRange makes more sense